src/Traits/Buildable.php

```diff
@@ -292,4 +292,19 @@ function () use ($arguments, $cacheKey, $method) {
                 }
             );
     }
+
+    /**
+     * Apply the scopes if they haven't already been applied, if they have
+     * just return the builder. This prevents scopes from being applied twice.
+     *
+     * @return static
+     */
+    public function applyScopes()
+    {
+        if ($this->scopesAreApplied) {
```

Comment:
@saernz  Quick question: where is the defined or updated when scopes are applied? I couldn't find it when doing a global search.

Reply:
Oh, nevermind, I found it in Caching.php ... sorry, overlooked that earlier.